src/test/java/uk/gov/pay/products/resources/ProductsResourceTest.java

```diff
@@ -217,5 +217,60 @@ public void givenANotAuthenticatedRequest_shouldReturn401() throws Exception {
                 .get(format("/v1/api/products/%s", randomUuid()))
                 .then()
                 .statusCode(401);
+
+        givenSetup()
+                .when()
+                .accept(APPLICATION_JSON)
+                .get(format("/v1/api/products?externalServiceId=%s", randomUuid()))
+                .then()
+                .statusCode(401);
+    }
+
+    @Test
+    public void givenAnExistingExternalServiceId_shouldFindAndReturnProducts() throws Exception {
+        String externalServiceId = randomUuid();
+        CatalogueEntity aCatalogueEntity = aCatalogueEntity()
+                .withExternalServiceId(externalServiceId)
+                .build();
+
+        Product product = ProductEntityFixture.aProductEntity()
+                .withCatalogue(aCatalogueEntity)
+                .build()
+                .toProduct();
+
+        int catalogueId = randomInt();
+        databaseHelper.addProductAndCatalogue(product, catalogueId);
+
+        Product product_2 = ProductEntityFixture.aProductEntity()
+                .withCatalogue(aCatalogueEntity)
+                .build()
+                .toProduct();
+
+        databaseHelper.addProduct(product_2, catalogueId);
+
+        ValidatableResponse response = givenAuthenticatedSetup()
+                .when()
+                .accept(APPLICATION_JSON)
+                .get(format("/v1/api/products?externalServiceId=%s", externalServiceId))
+                .then()
+                .statusCode(200);
+
+        response.body("", hasSize(2))
+                .body("[0].external_service_id", matchesPattern(externalServiceId))
+                .body("[0]._links", hasSize(2))
+                .body("[0].description", matchesPattern(product.getDescription()))
+                .body("[0].price", is(product.getPrice().intValue()))
+                .body("[0].name", matchesPattern(product.getName()))
+                .body("[1].name", matchesPattern(product_2.getName()));
```

Comment:
why are we not checking all things attributes of product 2?

Reply:
Given that the above list is generated using a recursive call on a method that was tested earlier to generate the expected output I considered not testing for the second product as the functionality that will generate the values was tested above.

Reply:
fair enough 👍